### ISSUE TITLE
`1.2.2` breaks `sprockets-rails` `2.x`?

### ISSUE THREAD

**Issue by PikachuEXE** [railsware/js-routes#177]
# Env
- `sprockets-rails (2.3.3)`
- `sprockets (2.12.4)`
- `rails (4.1.14)`

# Error
`TypeError: can't modify immutable index`

# Backtrace
```
backtrace: [GEM_ROOT]/gems/sprockets-2.12.4/lib/sprockets/index.rb:81:in `expire_index!'
[GEM_ROOT]/gems/sprockets-2.12.4/lib/sprockets/base.rb:175:in `register_preprocessor'
[GEM_ROOT]/gems/js-routes-1.2.2/lib/js_routes/engine.rb:11:in `block in <class:Engine>'
[GEM_ROOT]/gems/activesupport-4.1.14/lib/active_support/lazy_load_hooks.rb:36:in `call'
[GEM_ROOT]/gems/activesupport-4.1.14/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
[GEM_ROOT]/gems/activesupport-4.1.14/lib/active_support/lazy_load_hooks.rb:45:in `block in run_load_hooks'
[GEM_ROOT]/gems/activesupport-4.1.14/lib/active_support/lazy_load_hooks.rb:44:in `each'
[GEM_ROOT]/gems/activesupport-4.1.14/lib/active_support/lazy_load_hooks.rb:44:in `run_load_hooks'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/application/finisher.rb:64:in `block in <module:Finisher>'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/initializable.rb:30:in `instance_exec'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/initializable.rb:30:in `run'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/initializable.rb:55:in `block in run_initializers'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:345:in `each'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:345:in `call'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
/Users/PikachuEXE/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/initializable.rb:54:in `run_initializers'
[GEM_ROOT]/gems/railties-4.1.14/lib/rails/application.rb:300:in `initialize!'
[PROJECT_ROOT]/config/environment.rb:5:in `<top (required)>'
(More but not I guess not related)
```

**Comment by bogdan** [comment]
And what is the error message?

**Comment by PikachuEXE** [comment]
Sorry, just added to the issue description

**Comment by le0pard** [comment]
Maybe we just need back to `initializer 'js-routes.dependent_on_routes'` block, because it was changed to `config.after_initialize` block.

**Comment by drakmail** [comment]
:+1: Same issue, but reproduces only when compiling assets with `RAILS_ENV=production`

**Comment by bogdan** [comment]
@PikachuEXE can you try current master and check if it fixed your problem?

**Comment by nimashariatian** [comment]
@bogdan  i can confirm  b713dcc fixes the issue

# Env 

* sprockets-rails (2.3.3)
* sprockets (2.12.4)
* rails (4.2.5.1)

**Comment by bogdan** [comment]
@nimashariatian I'll wait for other people to try and release soon.

**Comment by nimashariatian** [comment]
thank you!

**Comment by PikachuEXE** [comment]
@bogdan 
I can confirm that b713dcc does fix the issue

## Env
# Env
- `sprockets-rails (2.3.3)`
- `sprockets (2.12.4)`
- `rails (4.1.14.1)`

**Comment by bogdan** [comment]
Released v1.2.3
